### ISSUE TITLE
Feature request for multitenant(nested organization)

### ISSUE THREAD

**Issue by ranjithkodumbu** [NagiosEnterprises/nagioscore#676]
Hi Team,
Currently we do not have real multitenant environment, we can achieve partially by creating contact name as a user and attach particular contact in all the host/services. 

But real multitenant requirement is different, something like we should be able to create organization, in that environment need to onboard host/services and better user management with different roles like admin/readonly etc and also should be able to create sub-organization etc. 

Only super admin should have full access to the all organization by default.

Thanks, 
Ranjith Kumar R

**Comment by ericloyd** [comment]
We have long been a proponent of this very thing.  However, given the current development cycle of Nagios and state of the code as it stands today, it's going to take a major rewrite of the underlying object model before this can happen.  I do know it's on the roadmap, however.

**Comment by Madlohe** [comment]
Hi @ranjithkodumbu,

You might want to look into using Nagios XI. It's a paid product, but it addresses most of the issues you bring up here. Given the current architecture of Core, it's very unlikely that multitenancy and similar features would be added anytime soon.

**Comment by ericloyd** [comment]
To be fair though, XI isn't quite the same thing, either.  There is no "Account management" for a sub-account within XI.  Multi-tenancy is still based on contacts and notifications and if you have "modify all" privs, you can modify everything, not just the stuff in your "organization."  So I think @ranjithkodumbu still has valid points about Nagios (as a whole) lacking true managed organization based multi-tenancy.

Think root (controls everything), organization admin (Network organization and Applications organization, each of which has their own admins but are separate from each other), department admin (App QA versus App Production for instance), and users of all sorts, and XI fails this multi-level administrative capability.

**Comment by ranjithkodumbu** [comment]
You are right Ericloyd, my point also same. I am using Nagiosxi, but it is same as Nagios core in perspective of user management. And it doesn't have real multitenancy.
